### DISCUSSION TITLE
classes added via AddUtilities functions are not compiled in JIT Mode

### DISCUSSION THREAD

**Discussion by AndreaChiabotto** [tailwindlabs/tailwindcss#4785] [Help]
### What version of Tailwind CSS are you using?

2.2.0

### What build tool (or framework if it abstracts the build tool) are you using?

Tailwind Play

### What version of Node.js are you using?

Tailwind Play

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### Reproduction repository

https://play.tailwindcss.com/xGuk7gk8ds?layout=horizontal

### Describe your issue

I have a grid system that is flex based and with the old mode everything works fine.
If I add 'mode:jit', columns are skipped, or not compiled. I think this happens because I set the variant 'responsive' during the function or the name ( to stick consistent to the Tailwind system) set the name of the classes like the breakpoints. I had the same problem on my Project ( Angular 12 ) but it doesn't work either on tailwind play.

**Comment by adamwathan** [comment]
Hey! I don't fully understand what's happening in your plugin at a quick glance but it looks like it's doing way more work than I'd expect. What's the reason for generating all the variant classes yourself like `s:col-2`? Tailwind will generate those for you, that's the idea behind having responsive variants. You only need to add the `col-2` class, Tailwind will create the `s:col-2` class for you automatically.

Most of the rules in there don't really look like utilities to me, it looks like a lot of complex CSS with nested media queries and stuff. I wonder if maybe this custom CSS is a bad fit for what the plugin system is meant for?

**Comment by adamwathan** [comment]
Going to move this to a discussion as the plugin system is definitely not meant to be used to create classes that already have variants baked into the class name, Tailwind generates all the variants for you just given the basic class names with no variants in them.

If you include the variants yourself Tailwind is going to do weird things like try to double them up, giving you classes like `sm:sm:col-2` and `lg:sm:col-2` because it tries to create variants of classes that already have variants in them.

Can definitely discuss how to achieve what you want but I don't see this as a bug, so discussion is a better spot.

**Comment by AndreaChiabotto** [comment]
HI Adam, thank you for the really fast answer! I really appreciated that. :)

For the Plugin, is intended to work as a mixin:
I need some classes to operate responsive, and for this I used the variant. (.col, .rows
For others, I don't need the responsiveness via tailwind, because the columns behave different on different breakpoint (I need only 2 columns on Mobile, 6 columns on tablet and 12 on desktop). 

The naming of the classes, starting with the breakpoint ( xs,s an l) is there because we wanted that the classes pop up during development, and the intellij plugin help us delivering them. Of course, we can rename these classes with other prefix, but we wanted to be consistent with the Tailwind logic.

I needed to add this logic into the Tailwind config, because it's the core part of our design library and therefore we didn't want to outside into a separate css file.

I hope that i could clarify the process a little bit.
Help us Adam, you are our only hope...

**Reply by AndreaChiabotto** [reply]
Hallo, I have tested something more and it seems that the JIT mode simply take out the classes that are generated via addUtilities. There is no variant Anymore, the classes are simply not showing, therefore i think is a bug.

https://play.tailwindcss.com/CFVOrOOnMH?layout=horizontal
